Source/Initialization/WarpXInitData.cpp

```diff
@@ -1072,48 +1059,61 @@ WarpX::InitLevelData (int lev, Real /*time*/)
     }
 }
 
-void
-WarpX::InitializeExternalFieldsOnGridUsingParser (
-       MultiFab *mfx, MultiFab *mfy, MultiFab *mfz,
-       ParserExecutor<3> const& xfield_parser, ParserExecutor<3> const& yfield_parser,
-       ParserExecutor<3> const& zfield_parser,
-       ablastr::fields::VectorField const& edge_lengths,
-       ablastr::fields::VectorField const& face_areas,
-       [[maybe_unused]] const char field,
-       const int lev, PatchType patch_type)
+void WarpX::ComputeExternalFieldOnGridUsingParser (
+    warpx::fields::FieldType field,
+    amrex::ParserExecutor<4> const& fx_parser,
+    amrex::ParserExecutor<4> const& fy_parser,
+    amrex::ParserExecutor<4> const& fz_parser,
+    int lev, PatchType patch_type, [[maybe_unused]] const char topology,
+    std::optional<ablastr::fields::VectorField> const& edge_lengths,
+    std::optional<ablastr::fields::VectorField> const& face_areas)
 {
+    auto t = gett_new(lev);
 
     auto dx_lev = geom[lev].CellSizeArray();
-    amrex::IntVect refratio = (lev > 0 ) ? WarpX::RefRatio(lev-1) : amrex::IntVect(1);
+    const RealBox& real_box = geom[lev].ProbDomain();
+
+    amrex::IntVect refratio = (lev > 0 ) ? RefRatio(lev-1) : amrex::IntVect(1);
     if (patch_type == PatchType::coarse) {
         for (int idim = 0; idim < AMREX_SPACEDIM; ++idim) {
-            dx_lev[idim] = dx_lev[idim] * refratio[idim];
+            dx_lev[idim] = dx_lev[idim] + refratio[idim];
```

Comment:
I'm guessing this is why the CI test failed:
```suggestion
            dx_lev[idim] = dx_lev[idim] * refratio[idim];
```

Reply:
Yea typo from trying to fix clang-tidy problems but I haven't gotten it to compile yet. Maybe I'm missing it but I can't see the errors from the azure pipeline.

Reply:
Yeah the clang-tidy errors are quite hard to spot since they get buried in lots of other output. My trick is to search for the carrot symbol (`^`) in the output since usually the errors have that symbol.

Reply:
Keep in mind that our `clang-tidy` builds run through GitHub Actions, not Azure. If you scroll down to the bottom of the main page of this PR, where all CI checks are listed, you can click on "Details" next to a given `clang-tidy` check (for example, the workflow labeled `clang-tidy / clang-tidy-1D (pull_request)`) and you should be redirected to a page where you can see the output of all the steps of that particular `clang-tidy` workflow, which in turn will display build errors, if any. Do you manage to access the output logs like this?

Reply:
[EZoni](https://github.com/EZoni) You are right thanks, I missed that it was GitHub Actions. I have access to it so hopefully things will go much smoother.